tsdb/engine/tsm1/engine.go

```diff
@@ -2134,7 +2134,12 @@ func (e *Engine) compact(wg *sync.WaitGroup) {
 			// If no full compactions are need, see if an optimize is needed
 			var genLen int64
 			if len(level4Groups) == 0 {
-				level4Groups, len4, genLen = e.CompactionPlan.PlanOptimize()
+				level4Groups, len4, genLen = e.CompactionPlan.PlanOptimize(e.LastModified())
+				if len(level4Groups) > 0 {
+					for _, group := range level4Groups {
+						e.logger.Info("TSM scheduled for optimized compaction", zap.Any("file", group))
```

Comment:
What does the `zap.Any` for the `group` print in the log?  This looks like it could be a huge log line.

Maybe make a `tsmGeneration` method that returns a slice with just the file names, and use `zap.StringArray` or whatever?

Reply:
`group` is a list of the TSM files that will be picked up during compaction:

```go
type CompactionGroup []string
```

I could not use `zap.StringArray` with it. Likely because its a custom type? I'll try to modify this.

Reply:
Disregard I was able to use `zap.Strings`